src/fpylll/algorithms/bkz2.py

```diff
@@ -17,13 +17,18 @@ def __init__(self, A):
         """
         BKZBase.__init__(self, A)
 
-    def get_pruning(self, kappa, block_size, param, tracer=dummy_tracer):
-        strategy = param.strategies[block_size]
+    def get_pruning(self, kappa, block_size, params, tracer=dummy_tracer):
+        strategy = params.strategies[block_size]
+        radius = self.M.get_r(kappa, kappa) * self.lll_obj.delta
 
-        radius, re = self.M.get_r_exp(kappa, kappa)
-        root_det = self.M.get_root_det(kappa, kappa + block_size)
-        gh_radius, ge = adjust_radius_to_gh_bound(radius, re, block_size, root_det, 1.0)
-        return strategy.get_pruning(radius  * 2**re, gh_radius * 2**ge)
+        # Don't bother computing expensive gh for small blacksizes
+        if not (params.flags & BKZ.GH_BND and block_size > 30):
+            return radius, strategy.get_pruning(radius, 100 * radius)
```

Comment:
You are not daft. Indeed, I only meant that for block-size < 30. It must be ammended. Maybe I can check the effect of always doing it (ie computing the gh heuristic) and see if there is no slowdown...